src/sharedHooks/useSuggestions/useOfflineSuggestions.ts

```diff
@@ -67,30 +62,21 @@ const useOfflineSuggestions = (
     // similar to what we're doing in the AICamera to get iconic taxon name,
     // but we're offline so we only need the local list from realm
     // and don't need to fetch taxon from the API
-    const iconicTaxa = realm?.objects( "Taxon" ).filtered( "isIconic = true" );
-    const iconicTaxaIds = iconicTaxa.map( t => t.id );
-    const iconicTaxaLookup: {
-      [key: number]: string;
-    } = iconicTaxa.reduce( ( acc, t ) => {
-      acc[t.id] = t.name;
-      return acc;
-    }, { } );
```

Comment:
Change on the right fixes the _last_ type error in `sharedHooks/useSuggestions`. I found the pattern https://github.com/inaturalist/iNaturalistReactNative/blob/1411a9a5238f85a83661adb9f61b40b457ddff0e/src/components/MyObservations/MyObservationsContainer.tsx#L278-L279 we use for type info from `realm.object`. Using `Object.entries` and `in` [below here](https://github.com/inaturalist/iNaturalistReactNative/pull/3257/changes#diff-059a5fd6a2ae97e57d7407fd2d7c501a159177d4af8a782f232e4397d214c688R79) reduces the amount of type definition we have to do.